orttraining/orttraining/core/graph/training_op_defs.cc

```diff
@@ -1047,6 +1047,155 @@ void RegisterTrainingOpSchemas() {
+          "weights",
+          "Sequence of weights to optimize.",
+          "S_WEIGHT")
+      .Input(
+          3,
+          "gradients",
+          "Sequence of gradients computed in this iteration.",
+          "S_GRAD")
+      .Input(
+          4,
+          "moment_1",
+          "Sequence of exponentially averaged historical gradients.",
+          "S")
+      .Input(
+          5,
+          "moment_2",
+          "Sequence of exponentially averaged historical squared gradients.",
+          "S")
+      .Output(
+          0,
+          "is_updated",
+          "Whether gradient is applied or not, mostly used as the optimizer applying trigger.",
+          "T2")
+      .Attr(
+          "alpha",
+          "Coefficient of previous gradient in running average.",
+          AttributeProto::FLOAT,
+          0.9f)
+      .Attr(
+          "beta",
+          "Coefficient of previous squared gradient in running average."
+          "The effective learning rate is computed by r = R / (1 + T * decay_factor). "
+          "Default to 0 so that increasing update counts doesn't reduce the learning rate.",
+          AttributeProto::FLOAT,
+          0.999f)
+      .Attr(
+          "lambda",
+          "Regularization coefficient of 0.5 * lambda * ||X||_2^2. Default to 0, "
+          "which means no regularization.",
+          AttributeProto::FLOAT,
+          0.0f)
+      .Attr(
+          "epsilon",
+          "Small scalar to avoid dividing by zero.",
+          AttributeProto::FLOAT,
+          1e-8f)
+      .Attr(
+          "do_bias_correction",
+          "Compute unbiased 1st and 2nd momentums.",
+          AttributeProto::INT,
+          static_cast<int64_t>(1))
+      .Attr(
+          "weight_decay_mode",
+          "Modes for applying weight decay, "
+          "0 means applying decay before weight update, "
+          "1 means applying decay after weight update.",
+          AttributeProto::INT,
+          static_cast<int64_t>(0))
+      .TypeConstraint(
+          "S",
+          {"seq(tensor(float16))", "seq(tensor(float))", "seq(tensor(double))", "seq(tensor(bfloat16))"},
+          "Constrain input types to float tensors")
+      .TypeConstraint(
+          "S_WEIGHT",
+          {"seq(tensor(float16))", "seq(tensor(float))", "seq(tensor(double))"},
+          "Constrain input types to float tensors.")
+      .TypeConstraint(
+          "S_GRAD",
+          {"seq(tensor(float16))", "seq(tensor(float))", "seq(tensor(double))", "seq(tensor(bfloat16))"},
+          "Constrain input types to float tensors.")
+      .TypeConstraint(
+          "T1",
+          {"tensor(float16)", "tensor(float)", "tensor(double)", "tensor(bfloat16)"},
+          "Constrain learning rate to float")
+      .TypeConstraint(
+          "T2",
+          {"int64"},
+          "Constrain step count to 64-bit integer");
+
+  ONNX_CONTRIB_OPERATOR_SCHEMA(InPlaceToSequence)
+      .SetDomain(kMSDomain)
+      .SinceVersion(1)
+      .SetDoc("")
```

Comment:
Add description for the op.

Reply:
How is this different than SequenceConstruct? https://github.com/onnx/onnx/blob/main/docs/Operators.md#sequenceconstruct

Reply:
It is trying the reuse all inputs' memory (OrtValues), the dimplementation of SequenceConstruct in current ORT is doing copy. We need optimizer in-place operater the weight update